File: protocol/python/src/lokascript_explicit/references.py
"""Default reference set for LSE."""

DEFAULT_REFERENCES: frozenset[str] = frozenset(
    {"me", "you", "it", "result", "event", "target", "body"}
)


def is_valid_reference(
    value: str, reference_set: frozenset[str] | set[str] | None = None
) -> bool:
    """Check if a value is a valid reference name."""
    refs = reference_set if reference_set is not None else DEFAULT_REFERENCES
    return value in refs

File: protocol/rust/src/references.rs
//! Default reference set for LSE.

use std::collections::HashSet;

/// Returns the default set of valid reference names.
pub fn default_references() -> HashSet<String> {
    ["me", "you", "it", "result", "event", "target", "body"]
        .iter()
        .map(|s| s.to_string())
        .collect()
}

/// Check if a value is a valid reference name (case-insensitive).
pub fn is_valid_reference(value: &str, reference_set: &HashSet<String>) -> bool {
    reference_set.contains(&value.to_lowercase())
}
